http-common/src/connector.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Connector {
+	Http { host: std::sync::Arc<str>, port: u16 },
+	Unix { socket_path: std::sync::Arc<std::path::Path> },
+}
+
+#[derive(Debug)]
+pub enum Stream {
+	Http(std::net::TcpStream),
+	Unix(std::os::unix::net::UnixStream),
+}
+
+#[derive(Debug)]
+pub enum AsyncStream {
+	Http(tokio::net::TcpStream),
+	Unix(tokio::net::UnixStream),
+}
+
+#[derive(Debug)]
+pub enum Incoming {
+	Http(tokio::net::TcpListener),
+	Unix(tokio::net::UnixListener),
+}
+
+impl Connector {
+	pub fn new(uri: &url::Url) -> Result<Self, ConnectorError> {
+		match uri.scheme() {
+			"http" => {
```

Comment:
Can we put a test feature flag to enable http? I don't think we need to support http by default after UDS is enabled. 
This also reminds me - I should rip out the https protocol config in identityd. I can handle that in my UDS PR for identityd though.

Reply:
`http` is needed because `iotedge` uses reqwest to invoke it directly, which in turn is because there's no `Connector`-based client for IS. That's why I left the IS docs and default config to still use `http`. If you're going to add a `Connector`-based client in your PR, then you can remove `http` entirely in that PR.

Reply:
I was initially thinking we could use it for E2E tests, but that isn't useful either. I can remove it after IS gets UDS. Thanks.

Reply:
Yes, E2E can use UDS too.

We will need http/https eventually for Kubernetes, but let's worry about that when that happens.